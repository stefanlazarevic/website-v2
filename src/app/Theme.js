export const light = {
  primary: '#8268ff',
  secondary: '#009bec',
  danger: '#ff3b30',
  success: '#28cd41',
  info: '#007aff',
  warning: '#ffcc00',
  background: {
    primary: '#fff',
    secondary: '#274c7e0d',
    disabled: '#fefefe',
  },
  text: {
    primary: '#343d46',
    secondary: '#57585a',
  },
  border: {
    primary: '#00000012',
  },
  link: {
    primary: '#8268ff',
  },
};

export const dark = {
  primary: '#8268ff',
  secondary: '#009bec',
  danger: '#ff453a',
  success: '#32d74b',
  info: '#0a84ff',
  warning: '#ffd60a',
  background: {
    primary: '#171a1f',
    secondary: '#2E3541',
    disabled: '#1F2229',
  },
  text: {
    primary: '#fff',
    secondary: '#c2c2c2',
  },
  border: {
    primary: '#434c5c',
  },
  link: {
    primary: '#8268ff',
  },
};

export default light;
