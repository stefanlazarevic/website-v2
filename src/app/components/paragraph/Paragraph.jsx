import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const P = styled.p`
  font-family: 'Nunito', sans-serif;
  font-size: 20px;
  line-height: 34px; /* (1.7rem) */
  margin: 0 0 30px; /* 0 0 1.5rem */
  color: ${props => props.theme.text.secondary};
  font-weight: 300;
`;

class Paragraph extends PureComponent {
  render = () => <P {...this.props}>{this.props.children}</P>;
}

Paragraph.propTypes = {
  children: PropTypes.node,
};

export default Paragraph;
