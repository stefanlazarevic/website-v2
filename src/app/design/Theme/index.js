import Typography from '../Typography';
import { LightSwatch, DarkSwatch } from '../Color';

export const LightTheme = Object.assign({}, LightSwatch, Typography);
export const DarkTheme = Object.assign({}, DarkSwatch, Typography);

export default LightTheme;
