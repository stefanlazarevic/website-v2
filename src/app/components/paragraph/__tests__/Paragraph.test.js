import React from 'react';
import Paragraph from '../Paragraph';
import { shallow, mount } from 'enzyme';
import { light } from '../../../Theme';

describe('Paragraph component', () => {
  it('Render paragraph element.', () => {
    const paragraph = shallow(<Paragraph />);
    expect(paragraph).toMatchSnapshot();
  });

  it('Render text inside paragraph,', () => {
    const paragraph = mount(<Paragraph theme={light}>Hello World</Paragraph>);
    expect(paragraph.text()).toEqual('Hello World');
  });

  it('Render child components inside paragraph.', () => {
    const paragraph = mount(
      <Paragraph theme={light}>
        This text is <strong>bold.</strong>
      </Paragraph>
    );

    expect(paragraph.text()).toEqual('This text is bold.');
  });
});
