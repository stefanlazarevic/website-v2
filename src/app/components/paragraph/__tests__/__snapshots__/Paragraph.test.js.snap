// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph component Render paragraph element. 1`] = `<Paragraph__P />`;
